src/Features/CSharp/Portable/EditAndContinue/TopSyntaxComparer.cs

```diff
@@ -346,7 +347,16 @@ public override bool ValuesEqual(SyntaxNode left, SyntaxNode right)
                 case SyntaxKind.RemoveAccessorDeclaration:
                     // When comparing method bodies we need to NOT ignore VariableDeclaration and VariableDeclarator children,
                     // but when comparing field definitions we should ignore VariableDeclarations children.
-                    ignoreChildFunction = childKind => HasLabel(childKind, ignoreVariableDeclarations: true);
+
+                    var leftBody = GetBody(left);
+                    var rightBody = GetBody(right);
+                    
+                    if (!SyntaxFactory.AreEquivalent(leftBody, rightBody, null))
+                    {
+                        return false;
+                    }
+
+                    ignoreChildFunction = childKind => childKind == SyntaxKind.Block || childKind == SyntaxKind.ArrowExpressionClause || HasLabel(childKind, ignoreVariableDeclarations: true);
```

Comment:
OK. Thanks! Let us try

---
In reply to: [133794959](https://github.com/dotnet/roslyn/pull/21573#discussion_r133794959) [](ancestors = 133794959)